cognite/client/data_classes/data_modeling/ids.py

```diff
@@ -69,15 +69,55 @@ def load(
 T_Versioned_DataModeling_Id = TypeVar("T_Versioned_DataModeling_Id", bound=VersionedDataModelingId)
 
 
+@dataclass
+class InstanceId:
+    _instance_type: ClassVar[str]
+    space: str
+    external_id: str
+
+    def dump(self, camel_case: bool = False, include_instance_type: bool = True) -> dict[str, str]:
+        output = asdict(self)
+        if include_instance_type:
+            output["instanceType" if camel_case else "instance_type"] = self._instance_type
+        return convert_all_keys_recursive(output, camel_case)
+
+    @classmethod
+    def load(cls: Type[T_InstanceId], data: dict) -> T_InstanceId:
+        return cls(
+            **convert_all_keys_to_snake_case(rename_and_exclude_keys(data, exclude={"instanceType", "instance_type"}))
+        )
+
+
+T_InstanceId = TypeVar("T_InstanceId", bound=InstanceId)
+
+
+@dataclass
+class NodeId(InstanceId):
+    _instance_type = "node"
+
+
+@dataclass
+class EdgeId(InstanceId):
+    _instance_type = "edge"
+
+
 @dataclass
 class ContainerId(DataModelingId):
     _type = "container"
 
+    @property
+    def identifier(self) -> str:
+        raise NotImplementedError
+
 
 @dataclass
 class ViewId(VersionedDataModelingId):
     _type = "view"
 
+    @property
+    def identifier(self) -> str:
+        return f"{self.external_id}/{self.version}"
```

Comment:
Hmm. Not so sure about these `identifier` properties on ID objects. Not very intuitive to me what that would return. Perhaps drop the property on ContainerId and just be super explicit in the ViewId naming. smth like 
```
@property
def external_id_slash_version(self) -> str:
    return f"{self.external_id}/{self.version}"
```

Reply:
Or I see it's being used to created property references. So perhaps have something like this instead:

```
class PropertyRef:
    @overload
    @staticmethod
    def of(view_id: ViewId, property_identifier: str) -> tuple[str, str, str]:
        ...

    @overload
    @staticmethod
    def of(container_id: ContainerId, property_identifier: str) -> tuple[str, str, str]:
        ...

    @staticmethod
    def of(view_or_container_id: ContainerId | ViewId, property_identifier: str) -> tuple[str, str, str]:
        if isinstance(view_or_container_id, ViewId):
            external_id_maybe_versioned = f"{view_or_container_id.external_id}/{view_or_container_id.version}"
        elif isinstance(view_or_container_id, ContainerId):
            external_id_maybe_versioned = view_or_container_id.external_id
        else:
            raise ValueError
        return (view_or_container_id.space, external_id_maybe_versioned, property_identifier)

```

Reply:
What about renaming it and making it a method? As far as I can see the PropertyRef.of is not the same as what I need.

```python
@dataclass
class ViewId(VersionedDataModelingId):
    _type = "view"

    def as_property_identifier(self) -> str:
        return f"{self.external_id}/{self.version}"
```